### PR TITLE
@DataMybatisTest similar to @DataJpaTest

### PR DESCRIPTION
**Pull request by wonwoo**
Hello everybody,

this @DataMybatisTest similar to @DataJpaTest.
You can easily test Mybatis.

```java
@RunWith(SpringRunner.class)
@DataMybatisTest
public class SampleDataMybatisTests {

  @Autowired
  private SampleMapper sampleMapper;

  @Autowired
  private SqlSession sqlSession;

  @Test
  public void sqlSessionIsNotNullTest() {
    assertThat(sqlSession).isNotNull();
  }

  @Test
  public void mapperIsNotNullTest() {
    assertThat(sampleMapper).isNotNull();
  }
}
```
are you okay?

Thank you in advance.

### CODE REVIEW

**Comment by eddumelendez** [comment]
Thanks for the PR @wonwoo it is great! I will take a look

**Comment by eddumelendez** [comment]
@wonwoo can you add new tests to the existing modules in `mybatis-spring-boot-samples`. Also, I think `MybatisTest` would be much better name. Thanks again.

File: mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/AutoConfigureDataMybatis.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ *
+ * @author wonwoo
+ * @since 1.2.1
+ */
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Inherited
+@ImportAutoConfiguration
+public @interface AutoConfigureDataMybatis {
```
Comment (eddumelendez) [inline mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/AutoConfigureDataMybatis.java]: Rename to `AutoConfigureMybatis`

File: mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/DataMybatisTest.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+@Inherited
+@BootstrapWith(SpringBootTestContextBootstrapper.class)
+@OverrideAutoConfiguration(enabled = false)
+@TypeExcludeFilters(DataMybatisTypeExcludeFilter.class)
+@AutoConfigureCache
+@Transactional
+@AutoConfigureDataMybatis
+public @interface DataMybatisTest {
```
Comment (eddumelendez) [inline mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/DataMybatisTest.java]: Rename to `MybatisTest`

File: mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/DataMybatisTest.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import org.springframework.context.annotation.ComponentScan.Filter;
+import org.springframework.test.context.BootstrapWith;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.lang.annotation.*;
+
+/**
+ *
+ * @author wonwoo
+ * note : spring boot 1.5 add method excludeAutoConfiguration()
```
Comment (eddumelendez) [inline mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/DataMybatisTest.java]: remove this note

File: mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/DataMybatisTypeExcludeFilter.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import org.springframework.boot.test.autoconfigure.filter.AnnotationCustomizableTypeExcludeFilter;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * {@link TypeExcludeFilter} for {@link DataMybatisTest @DataMybatisTest}.
+ * note : Spring boot 1.5 add method getComponentIncludes()
```
Comment (eddumelendez) [inline mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/DataMybatisTypeExcludeFilter.java]: remove this note

File: mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/DataMybatisTypeExcludeFilter.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+ * @author wonwoo
+ * @since 1.2.1
+ */
+class DataMybatisTypeExcludeFilter extends AnnotationCustomizableTypeExcludeFilter {
```
Comment (eddumelendez) [inline mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/DataMybatisTypeExcludeFilter.java]: rename to `MybatisTypeExcludeFilter`

**Comment by wonwoo** [comment]
@eddumelendez rename class and remove note...  and.. created sample code

File: mybatis-spring-boot-samples/pom.xml [hunk 6]
```diff
@@ -29,5 +29,6 @@
   <modules>
     <module>mybatis-spring-boot-sample-annotation</module>
     <module>mybatis-spring-boot-sample-xml</module>
+    <module>mybatis-spring-boot-sample-test</module>
```
Comment (eddumelendez) [inline mybatis-spring-boot-samples/pom.xml]: new module is not necessary just add new test for the existing ones. Sorry to not be clear.

File: mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/MybatisTest.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.core.AutoConfigureCache;
+import org.springframework.boot.test.autoconfigure.filter.TypeExcludeFilters;
+import org.springframework.boot.test.context.SpringBootTestContextBootstrapper;
+import org.springframework.context.annotation.ComponentScan.Filter;
+import org.springframework.test.context.BootstrapWith;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.lang.annotation.*;
```
Comment (eddumelendez) [inline mybatis-spring-boot-test-autoconfigure/src/main/java/org/mybatis/spring/boot/test/autoconfigure/MybatisTest.java]: avoid import *

File: mybatis-spring-boot-test-autoconfigure/src/main/resources/META-INF/spring.factories [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# AutoConfigureDataMybatis auto-configuration imports
+org.mybatis.spring.boot.test.autoconfigure.AutoConfigureMybatis=\
+  org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration,\
```
Comment (eddumelendez) [inline mybatis-spring-boot-test-autoconfigure/src/main/resources/META-INF/spring.factories]: remove whitespace at the beginning

**Comment by wonwoo** [comment]
@eddumelendez remove test module and created test code in `mybatis-spring-boot-samples`

**Comment by eddumelendez** [comment]
Merged. Thanks!

**Comment by kazuki43zoo** [comment]
I've changed title and description.

**Comment by kazuki43zoo** [comment]
@eddumelendez 

Is not need to include `FlywayAutoConfiguration` and `LiquibaseAutoConfiguration` into `@AutoConfigureMybatis` ? 
The `@AutoConfigureDataJpa` include these auto-configure class.
I think these auto-configure class should be include into `@AutoConfigureMybatis`.
WDYT ?

> **Note:** 
> Incidentally ... The `@AutoConfigureJdbc` that added since 1.5 include these auto-configure class.

**Comment by kazuki43zoo** [comment]
ping @eddumelendez 
WDYT about above comment ?

**Comment by kazuki43zoo** [comment]
ping @eddumelendez 

WDYT about https://github.com/mybatis/spring-boot-starter/pull/143#issuecomment-285871759 ?

**Comment by eddumelendez** [comment]
Thanks for the reminder @kazuki43zoo check cfeeff4

**Comment by kazuki43zoo** [comment]
@eddumelendez Thanks for fix !!

**Comment by kazuki43zoo** [comment]
I've updated document via 7ac4b242a7c252b33e4fcd3d37b7b8e2ff7f60f7.

**Pull request closed without merging**